pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,413 @@
+        target_creators.dedup();
+
+        let current_era = Self::current_era();
+
+        for creator_id in target_creators {
+            let mut staker_info_for_creator = Self::staker_info(&staker, creator_id);
+
+            let mut unregistered_era = current_era;
+            if let Some(creator_info) = Self::registered_creator(creator_id) {
+                if let CreatorState::Unregistered(era) = creator_info.state {
+                    unregistered_era = era;
+                }
+            }
+
+            if staker_info_for_creator.stakes.is_empty() {
+                estimated_rewards.push((creator_id, Zero::zero()));
+                continue;
+            }
+
+            let mut total_staker_rewards_for_eras: BalanceOf<T> = Zero::zero();
+            loop {
+                let (era, staked) = staker_info_for_creator.claim();
+                if era >= unregistered_era || era == 0 {
+                    break;
+                }
+                let creator_stake_info = Self::creator_stake_info(creator_id, era).unwrap_or_default();
+
+                total_staker_rewards_for_eras = total_staker_rewards_for_eras.saturating_add(
+                    Self::calculate_reward_for_staker_in_era(&creator_stake_info, staked, era)
+                );
+            }
+
+            estimated_rewards.push((creator_id, total_staker_rewards_for_eras));
+        }
+
+        estimated_rewards
+    }
+
+    pub fn withdrawable_amounts_from_inactive_creators(
+        staker: T::AccountId,
+    ) -> Vec<(SpaceId, BalanceOf<T>)> {
+        let mut withdrawable_amounts_by_creator = Vec::new();
+
+        for (creator_id, staker_info) in GeneralStakerInfo::<T>::iter_prefix(&staker) {
+            if !Self::is_creator_active(creator_id) {
+                if let Some(most_recent_stake) = staker_info.stakes.last() {
+                    withdrawable_amounts_by_creator.push((creator_id, most_recent_stake.staked));
+                }
+            }
+        }
+
+        withdrawable_amounts_by_creator
+    }
+
+    pub fn available_claims_by_staker(
+        staker: T::AccountId,
+    ) -> Vec<(SpaceId, u32)> {
+        let mut available_claims_by_creator = BTreeMap::new();
+
+        let current_era = Self::current_era();
+
+        for (creator, mut stakes_info) in GeneralStakerInfo::<T>::iter_prefix(&staker) {
+            let mut unregistered_era = current_era;
+            if let Some(creator_info) = Self::registered_creator(creator) {
+                if let CreatorState::Unregistered(era) = creator_info.state {
+                    unregistered_era = era;
+                }
+            } else {
+                continue;
+            }
+
+            loop {
+                let (era, _) = stakes_info.claim();
+                if era >= unregistered_era || era == 0 {
+                    break;
+                }
+
+                available_claims_by_creator.entry(creator).and_modify(|e| *e += 1).or_insert(1);
```

Comment:
So here we are calculating how many times/eras we can claim. So if we have [(1, 5), (2, 10)] on response, on UI we will put claim for the creator #1 - 5 times and for #2 - 10 times, right?

Reply:
Pretty hard understand from name what does mean the `u32` in response. So maybe "available_claim_count_by_creator"